src/test/read_oversize.c

```diff
@@ -1,5 +1,9 @@
 /* -*- Mode: C; tab-width: 8; c-basic-offset: 2; indent-tabs-mode: nil; -*- */
 
+/* Disabling FORTIFY_SOURCE as the fortify checks cause RPM build failures. */
+#undef _FORTIFY_SOURCE
+#define _FORTIFY_SOURCE 0
+
```

Comment:
Can you work around this issue by replacing the `read` call with a `syscall(SYS_read, ...)` instead? If so I think I'd prefer that.

Reply:
Yeah, I think that would be possible. I'll try it out.

Reply:
This works, I'll make the changes.